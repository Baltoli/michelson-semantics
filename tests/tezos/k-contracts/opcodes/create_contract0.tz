$contract { parameter unit;
storage (option address);
code { DROP;
       UNIT; 
       AMOUNT;                   
       NONE key_hash;                 
       CREATE_CONTRACT          
         { parameter unit ;
           storage unit ;
           code
             { CDR;
               NIL operation;
               PAIR; } };
       DROP; SOME; NIL operation; PAIR} 
 } ;
$param Unit ;
$storage None ;
$ExpectedOutput (Some "KT1Mjjcb6tmSsLm7Cb3DSQszePjfchPM4Uxm") ;

