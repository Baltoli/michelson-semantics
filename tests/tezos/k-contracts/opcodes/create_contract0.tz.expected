<generatedTop>
  <k>
    #VerifyStorage ( Some #Address ( "KT1Mjjcb6tmSsLm7Cb3DSQszePjfchPM4Uxm" ) )
  </k>
  <stack>
    Pair .List Some #Address ( "0" )
  </stack>
  <paramtype>
    unit .AnnotationList
  </paramtype>
  <storagetype>
    option .AnnotationList address .AnnotationList
  </storagetype>
  <balance>
    #Mutez ( 0 )
  </balance>
  <amount>
    #Mutez ( 0 )
  </amount>
  <now>
    #Timestamp ( -1 )
  </now>
  <myaddr>
    #Address ( "InvalidMyAddr" )
  </myaddr>
  <knownaddrs>
    .Map
  </knownaddrs>
  <sourceaddr>
    #Address ( "InvalidSourceAddr" )
  </sourceaddr>
  <senderaddr>
    #Address ( "InvalidSenderAddr" )
  </senderaddr>
  <chainid>
    #ChainId ( 0 )
  </chainid>
</generatedTop>
